PusherMainView/PusherMainView/FCMPusher.swift

```diff
@@ -0,0 +1,183 @@
+import Foundation
+
+protocol FCMPushable {
+    func pushUsingLegacyEndpoint(_ token: String,
+                                 payload: [String: Any],
+                                 collapseID: String?,
+                                 serverKey: String,
+                                 completion: @escaping (Result<String, Error>) -> Void)
+    func pushUsingV1Endpoint(_ token: String,
+                             payload: [String: Any],
+                             collapseID: String?,
+                             serverKey: String,
+                             projectID: String,
+                             completion: @escaping (Result<String, Error>) -> Void)
+}
+
+class FCMPusher: FCMPushable {
+    let session = URLSession(configuration: .ephemeral,
+                             delegate: nil,
+                             delegateQueue: .main)
+
+    func pushUsingLegacyEndpoint(_ token: String,
+                                 payload: [String: Any],
+                                 collapseID: String?,
+                                 serverKey: String,
+                                 completion: @escaping (Result<String, Error>) -> Void) {
+
+        guard let url = URL(string: "https://fcm.googleapis.com/fcm/send") else {
+            completion(.failure(NSError(domain: "com.pusher.FCMPusher",
+                                        code: 0,
+                                        userInfo: [NSLocalizedDescriptionKey: "URL error"])))
+            return
+        }
+
+        var updatedPayload = payload
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("key=\(serverKey)", forHTTPHeaderField: "Authorization")
+
+        updatedPayload["to"] = token
+        if let collapseID = collapseID {
+            updatedPayload["collapse_key"] = collapseID
+        }
+
+        guard let httpBody = try? JSONSerialization.data(withJSONObject: payload, options: .prettyPrinted) else {
+            completion(.failure(NSError(domain: "com.pusher.FCMPusher",
+                                        code: 0,
+                                        userInfo: [NSLocalizedDescriptionKey: "Payload error"])))
+            return
+        }
+
+        request.httpBody = httpBody
```

Comment:
@imaginaris I think this code would be more readable if you move this URLRequest creation to an extension / helper.